### PR TITLE
Ssimreq

### PR DESCRIPTION
**Pull request by alexeilebedev**

### CODE REVIEW

File: cpp/acr/check.cpp [hunk 1]
```diff
@@ -157,6 +157,53 @@ static void CheckFunique() {
...
+        algo_lib::Regx regx;
+        Regx_ReadSql(regx,ssimreq.value,true);
+        ind_beg(acr::ctype_zd_selrec_curs, rec, *ssimreq.p_field->p_ctype) {
+            tempstr value(acr::EvalAttr(rec.tuple, *ssimreq.p_field));// find attribute value
+            if (Regx_Match(regx,value)) {
+                acr::FRec *childrec = ind_rec_Find(*ssimreq.p_ssimfile->p_ctype,rec.pkey);
+                // #AL# here it might be a good idea to actually create the missing record
+                // with defaults...
+                if (!childrec) {
+                    NoteErr(NULL,&rec,ssimreq.p_field,tempstr()<<"acr.ssimreq"
```
Comment (ali5h) [inline cpp/acr/check.cpp:172]: maybe issue an acr.insert instead?

File: data/dmmeta/field.ssim [hunk 2]
```diff
@@ -3106,6 +3111,11 @@ dmmeta.field  field:dmmeta.Ssimfile.ssimns  arg:dmmeta.Nsdb  reftype:Pkey  dflt:
 dmmeta.field  field:dmmeta.Ssimfile.ns  arg:dmmeta.Ns  reftype:Pkey  dflt:""  comment:""
 dmmeta.field  field:dmmeta.Ssimfile.name  arg:algo.Smallstr50  reftype:Val  dflt:""  comment:""
 dmmeta.field  field:dmmeta.Ssimfile.ctype  arg:dmmeta.Ctype  reftype:Pkey  dflt:""  comment:""
+dmmeta.field  field:dmmeta.Ssimreq.ssimfile  arg:dmmeta.Ssimfile  reftype:Pkey  dflt:""  comment:"Ssimfile with constraint"
+dmmeta.field  field:dmmeta.Ssimreq.field  arg:dmmeta.Field  reftype:Pkey  dflt:""  comment:"Field (must belong to pkey superset)"
+dmmeta.field  field:dmmeta.Ssimreq.value  arg:algo.Smallstr100  reftype:Val  dflt:""  comment:"Required value"
```
Comment (ali5h) [inline data/dmmeta/field.ssim:3116]: mention that this is a regx

File: cpp/acr/check.cpp (hunk 1 above)
Comment (alexeilebedev) [reply cpp/acr/check.cpp:172]: it's possible the child refers to parent via only a part of its pkey, meaning acr would have to guess the other part... cannot be solved in general case.

File: data/dmmeta/field.ssim (hunk 2 above)
Comment (alexeilebedev) [reply data/dmmeta/field.ssim]: done

**Review by ali5h — APPROVED** [review]

**Review by polovinkin-alexey — APPROVED** [review]

**Pull request merged**